tonic/models/vic/grid_params.py

```diff
@@ -1036,64 +1441,153 @@ def veg(veg_file, soil_dict, max_roots=3, veg_classes=11,
+            root_fract[cell, vind, :] = temp[3:1+tmp:2]
+            tmp += 1
 
-            else:
-                rind = (len(temp) - 5) / 2
-                sigma_slope[cell, vind, :rind] = temp[-3]
-                lag_one[cell, vind, :rind] = temp[-2]
-                fetch[cell, vind, :rind] = temp[-1]
+            if blowing_snow:
+                sigma_slope[cell, vind] = temp[tmp]
+                lag_one[cell, vind] = temp[1+tmp]
+                fetch[cell, vind] = temp[2+tmp]
+                tmp += 2
 
-            root_depth[cell, vind, :rind] = temp[2::2]
-            root_fract[cell, vind, :rind] = temp[3::2]
             row += 1
-            if lai_index:
+
+            if vegparam_lai:
                 lines[row] = lines[row].strip()
                 line = lines[row].strip('\n').split(' ')
                 lai[cell, vind, :] = np.array(line).astype(float)
                 row += 1
+            if vegparam_fcan:
+                lines[row] = lines[row].strip()
+                line = lines[row].strip('\n').split(' ')
+                fcan[cell, vind, :] = np.array(line).astype(float)
+                row += 1
+            if vegparam_albedo:
+                lines[row] = lines[row].strip()
+                line = lines[row].strip('\n').split(' ')
+                albedo[cell, vind, :] = np.array(line).astype(float)
+                row += 1
         cell += 1
     veg_dict = OrderedDict()
     veg_dict['gridcell'] = gridcel[:cell]
     veg_dict['Nveg'] = nveg[:cell]
     veg_dict['Cv'] = cv[:cell, :]
-    veg_dict['root_depth'] = root_depth[:cell, :]
-    veg_dict['root_fract'] = root_fract[:cell, :]
+    veg_dict['root_depth'] = root_depth[:cell, :, :]
+    veg_dict['root_fract'] = root_fract[:cell, :, :]
 
     if blowing_snow:
         veg_dict['sigma_slope'] = sigma_slope[:cell, :]
         veg_dict['lag_one'] = lag_one[:cell, :]
         veg_dict['fetch'] = fetch[:cell, :]
 
-    if lai_index:
+    if vegparam_lai and lai_src == FROM_VEGPARAM:
         veg_dict['LAI'] = lai[:cell, :, :]
 
+    if vegparam_fcan and fcan_src == FROM_VEGPARAM:
+        veg_dict['fcanopy'] = fcan[:cell, :, :]
+
+    if vegparam_albedo and alb_src == FROM_VEGPARAM:
+        veg_dict['albedo'] = albedo[:cell, :, :]
+
+    # Make gridcell order match that of soil_dict
     inds = []
     for sn in soil_dict['gridcell']:
         inds.append(np.nonzero(veg_dict['gridcell'] == sn))
 
+    new_veg_dict = OrderedDict()
     for var in veg_dict:
-        veg_dict[var] = np.squeeze(veg_dict[var][inds])
-    return veg_dict
+        new_veg_dict[var] = np.squeeze([veg_dict[var][i] for i in inds])
+
+    return new_veg_dict
+# -------------------------------------------------------------------- #
 
 
 # -------------------------------------------------------------------- #
-def veg_class(veg_file, maxcols=57, skiprows=3, c=Cols()):
+def lake(lake_file, soil_dict, max_numnod=10,
+        cells=False, lake_profile=False):
```

Comment:
I was referring just to the `cells` keyword argument in this case.  The reason is that `cells=False` is easily confused with `cells=0` or `cells=[]`.  Using `None` allows you to check the type below much easier.